File: sdk/src/main/java/com/meniga/sdk/webservices/requests/UpdateAccount.java
package com.meniga.sdk.webservices.requests;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public class UpdateAccount extends QueryRequestObject {

	public long id;
	public String name;
	public boolean isHidden;
	public boolean isDisabled;
	public Double emergencyFundBalanceLimit;

	@Override
	public long getValueHash() {
		int result = (int) (id ^ (id >>> 32));
		result = 31 * result + (name != null ? name.hashCode() : 0);
		result = 31 * result + (isHidden ? 1 : 0);
		result = 31 * result + (isDisabled ? 1 : 0);
		result = 31 * result + (emergencyFundBalanceLimit != null ? emergencyFundBalanceLimit.hashCode() : 0);
		return result;
	}
}

File: sdk/src/main/java/com/meniga/sdk/webservices/account/UpdateAccount.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.account

import com.meniga.sdk.webservices.requests.QueryRequestObject

data class UpdateAccount(
        @Transient var id: Long = 0,
        var name: String? = null,
        var orderId: Int = 0,
        var emergencyFundBalanceLimit: Double? = null,
        var isHidden: Boolean = false,
        var isDisabled: Boolean = false
) : QueryRequestObject() {

    override fun getValueHash() = hashCode().toLong()
}
